Bencodex.Tests/Misc/ImplicitConversionTest.cs

```diff
@@ -0,0 +1,71 @@
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Bencodex.Types;
+using Xunit;
+
+namespace Bencodex.Tests.Misc
+{
+    public class ConversionTest
+    {
+        [Fact]
+        public void Integer()
+        {
+            var integer = new Integer(1);
+
+            short s = integer;
+            ushort us = integer;
+            int i = integer;
+            uint ui = integer;
+            long l = integer;
+            ulong ul = integer;
+            Assert.Equal(s, integer.Value);
+            Assert.Equal(us, integer.Value);
+            Assert.Equal(i, integer.Value);
+            Assert.Equal(ui, integer.Value);
+            Assert.Equal(l, integer.Value);
+            Assert.Equal(ul, integer.Value);
```

Comment:
-  The first parameter of `Assert.Equal()` takes an expected value (not an actual value).
 -  Hard-coding literal 1 for expected values would be better.